packages/eslint-plugin/src/rules/return-await.ts

```diff
@@ -222,13 +250,29 @@ export default createRule({
         return;
       }
 
+      const affectsErrorHandling = affectsExplicitErrorHandling(expression);
```

Comment:
The thought is that one day there will be a `|| affectsExplicitResourceManagement(expression)` 🙏 https://github.com/typescript-eslint/typescript-eslint/issues/7889